app/src/main/java/com/skyyo/samples/features/textSpans/composables/UnderlineSpansText.kt

```diff
@@ -0,0 +1,238 @@
+                            )
+                        }
+                    drawPath(
+                        color = Color.Green,
+                        path = path,
+                        style = pathStyle
+                    )
+                }
+            }
+        }
+    )
+}
+
+@Composable
+fun SquiggleUnderlineText(
+    modifier: Modifier = Modifier,
+    text: AnnotatedString,
+    underlineWidth: TextUnit = 2.sp,
+    waveLength: TextUnit = 10.sp,
+    fontSize: TextUnit = TextUnit.Unspecified,
+    lineHeight: TextUnit = TextUnit.Unspecified
+) {
+    val density = LocalDensity.current
+    val onDraw: MutableState<DrawScope.() -> Unit> = remember { mutableStateOf({}) }
+    val pathStyle = Stroke(
+        width = with(density) { underlineWidth.toPx() },
+        join = StrokeJoin.Round,
+        cap = StrokeCap.Round,
+        pathEffect = PathEffect.cornerPathEffect(radius = with(density) { waveLength.toPx() }), // For slightly smoother waves.
+    )
+    Text(
+        modifier = modifier.drawBehind { onDraw.value(this) },
+        text = text,
+        fontSize = fontSize,
+        lineHeight = lineHeight,
+        onTextLayout = { layoutResult ->
+            val annotation = text.getStringAnnotations("squiggles", 0, text.length).first()
+            val textBounds = layoutResult.getBoundingBoxes(annotation.start, annotation.end)
+            onDraw.value = {
+                for (bound in textBounds) {
+                    val path = Path().apply { buildSquigglesFor(bound, density) }
+                    drawPath(
+                        color = Color.Green,
+                        path = path,
+                        style = pathStyle
+                    )
+                }
+            }
+        }
+    )
+}
+
+data class SimplePaddingValues(val horizontal: Dp, val vertical: Dp)
+
+@Composable
+fun HighlightedText(
+    modifier: Modifier = Modifier,
+    text: AnnotatedString,
+    fontSize: TextUnit = TextUnit.Unspecified,
+    lineHeight: TextUnit = TextUnit.Unspecified,
+    cornerRadius: CornerRadius,
+    highlightColor: Color,
+    highlightBorderColor: Color? = null,
+    highlightBorderWidth: Dp = 2.dp,
+    padding: SimplePaddingValues = remember {
+        SimplePaddingValues(
+            horizontal = 0.dp,
+            vertical = 0.dp
+        )
+    },
+) {
+    val onDraw: MutableState<DrawScope.() -> Unit> = remember { mutableStateOf({}) }
+    Text(
+        modifier = modifier.drawBehind { onDraw.value(this) },
+        text = text,
+        fontSize = fontSize,
+        lineHeight = lineHeight,
+        onTextLayout = { layoutResult ->
+            val annotation = text.getStringAnnotations("squiggles", 0, text.length).first()
+            val textBounds = layoutResult.getBoundingBoxes(annotation.start, annotation.end)
+            onDraw.value = {
+                textBounds.fastForEachIndexed { index, bound ->
+                    val path = Path().apply {
```

Comment:
instead of creating `Path` each time, we can move below to `val onDraw`, and do
`reset + addRoundRect`